### PR TITLE
support more dtype for Equal

### PR DESCRIPTION
**Pull request by lucienwang1009**
1. Support more dtypes for Equal ops and add some unit tests.
2. Move ops used to compare two tensors into [logical_compare.py](tf2onnx/function/logical_compare.py).

### CODE REVIEW

File: tf2onnx/function/logical_compare.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+"""
+tf2onnx.tf2onnx - logical compare ops
+"""
+from onnx.onnx_pb import TensorProto
+from tf2onnx import utils
+
+# pylint: disable=unused-argument,missing-docstring
+
+def add_cast_to_inputs(graph, node, supported_dtypes, target_dtype):
```
Comment (nbcsm) [inline tf2onnx/function/logical_compare.py]: make it internal.

**Comment by nbcsm** [comment]
please resolve conflict.

**Comment by lucienwang1009** [comment]
resolved:)

File: tf2onnx/onnx_opset/logical.py [hunk 2]
```diff
@@ -46,30 +41,56 @@ def version_4(cls, ctx, node, **kwargs):
...
 
+@tf_op("Equal")
+class Equal:
+    @classmethod
+    def version_4(cls, ctx, node, **kwargs):
+        common.BroadcastOp.version_4(ctx, node, **kwargs)
+
+    @classmethod
+    def version_7(cls, ctx, node, **kwargs):
+        # T2 output = Equal(T1, x, T1 y), T1 \in {bool, int32, int64}
```
Comment (nbcsm) [inline tf2onnx/onnx_opset/logical.py]: ```\```?

Comment (lucienwang1009) [reply tf2onnx/onnx_opset/logical.py]: aha \in means in_equation in math equations.

Comment (nbcsm) [reply tf2onnx/onnx_opset/logical.py]: learned new things. :-)

**Pull request merged**